### PR TITLE
Fixes missing analytics for the tab views

### PR DESCRIPTION
**Pull request by orta**
As this is testing the app delegate's launch code, I'm a bit at odds on how to add a test covering us for this. Going to take a stab at making it possible for us to run analytics tests, but I don't think I can cover _this case_.

### CODE REVIEW

**Comment by orta** [comment]
Alright, so, I've added the ability for us to test out that the analytics works. Ironically in the alanis morisette way, the first one I picked at random to test wasn't working. So I fixed that, and I've enforced the pattern via Danger.

File: Artsy_Tests/Stubs/ARStubbedAnalyticsProvider.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#import <ARAnalytics/ARAnalyticalProvider.h>
+#import <UIKit/UIKit.h>
+
+
```
Comment (orta) [inline Artsy_Tests/Stubs/ARStubbedAnalyticsProvider.h]: I just took this straight out of ARAnalytics, and added the ability to see an array of events incase we need more introspection

File: Dangerfile [hunk 2]
```diff
@@ -13,3 +13,8 @@ fail("fit left in tests") if `grep -r fit(@ Artsy_Tests/`.length > 1
 
 # Devs shouldn't ship changes to this file
 fail("Developer Specific file shouldn't be changed") if files_modified.include?("Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m")
+
+# Did you make analytics changes? Well you should also include a change to our analytics spec
+made_analytics_changes = files_modified.include?("/Artsy/App/ARAppDelegate+Analytics.m")
+made_analytics_specs_changes = files_modified.include?("/Artsy_Tests/Analytics_Tests/ARAppAnalyticsSpec.m")
+fail("Analytics changes should have reflected specs changes") if made_analytics_changes and !made_analytics_specs_changes
```
Comment (orta) [inline Dangerfile]: @KrauseFx I wonder if it's worth adding a DSL attribute around linking files e.g.

``` ruby
on_change "/Artsy/App/ARAppDelegate+Analytics.m", :expect => "/Artsy_Tests/Analytics_Tests/ARAppAnalyticsSpec.m"
```

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Error
    ------------- | ------------
        :no_entry_sign: | No CHANGELOG changes made
    

  

  
    &nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

File: Dangerfile (hunk 2 above)
Comment (KrauseFx) [inline Dangerfile]: That makes so much sense! I :heart: it. `expect:` should take a single string or an array, right?

File: Artsy/App/ARAppDelegate+Analytics.m [hunk 3]
```diff
@@ -92,10 +93,13 @@ - (void)setupAnalytics
         environment = ADJEnvironmentSandbox;
     }
 
-    // Skipping ARAnalytics because Adjust has its own expectations
-    // around event names being < 6 chars
-    ADJConfig *adjustConfig = [ADJConfig configWithAppToken:keys.adjustProductionAppToken environment:environment];
-    [Adjust appDidLaunch:adjustConfig];
+    if (ARAppStatus.isRunningTests == NO) {
```
Comment (alloy) [inline Artsy/App/ARAppDelegate+Analytics.m]: Just a small nitpick, but why not `!ARAppStatus.isRunningTests` ?

**Comment by alloy** [comment]
Can you please point-out thew specific code that fixes the tab views (e.g. bell tab view)?

**Comment by orta** [comment]
It was here: https://github.com/artsy/eigen/pull/1124/files#diff-41bc148e7a39dc3835f00c8ea76276cdR118

Analytics hooks into classes on alloc. The view structure was being set up before analytics were triggered. So the alloc was vanilla. Then any subsequent objects of whatever type ( in this case it was `ARTabContentView` ) would get the right analytics.

File: Dangerfile (hunk 2 above)
Comment (orta) [inline Dangerfile]: it's ruby, so both! :P

**Comment by alloy** [comment]
Aha, gotcha, thanks for the comment about that too :+1:

File: Dangerfile (hunk 2 above)
Comment (orta) [inline Dangerfile]: It’s Ruby,

-Insert Why Not Have Both GIF-

-- 
     [ ·/ ]  Orta Therox
w/  Artsy
CocoaPods /  CocoaDocs / GIFs
@orta  /  orta.io

On 5 February 2016 at 17:49:49, Felix Krause (notifications@github.com) wrote:

In Dangerfile:

> @@ -13,3 +13,8 @@ fail("fit left in tests") if `grep -r fit(@ Artsy_Tests/`.length > 1
>   
>  # Devs shouldn't ship changes to this file
>  fail("Developer Specific file shouldn't be changed") if files_modified.include?("Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m")
> +
> +# Did you make analytics changes? Well you should also include a change to our analytics spec
> +made_analytics_changes = files_modified.include?("/Artsy/App/ARAppDelegate+Analytics.m")
> +made_analytics_specs_changes = files_modified.include?("/Artsy_Tests/Analytics_Tests/ARAppAnalyticsSpec.m")
> +fail("Analytics changes should have reflected specs changes") if made_analytics_changes and !made_analytics_specs_changes
That makes so much sense! I  it. expect: should take a single string or an array, right?

**Pull request merged**